binderhub/app.py

```diff
@@ -250,6 +250,31 @@ def _add_slash(self, proposal):
         """
     )
 
+    use_as_federation_portal = Bool(False, config=True, 
+        help="""configure the current binder to also act as a federation portal.
+
+        This settings should be used only on mybinder.org deployment or if you
+        have multiple internal private deployment of binderhubs inside a closed
+        network. It exposes a couple of extra endpoints to binderhub for remote
+        services to register themselves. Once register for a specific user, the
+        binder portal will redirect users to these endpoints instead of running
+        them locally.
+        """
+    )
+
+    default_binders_list = List([], config=True,
+        help="""
+        A list of known default binder which can be chosen by users.
+        """
+    )
+
+    cannonical_address = Unicode('', config=True, 
+        help="""Set the canonical address to register self to a binder federation portal
```

Comment:
Mention if this needs a trailing slash or not and give an example?

Reply:
Done.